File: frontend/src/features/setup/data/costumeCatalog.js
/**
 * Costume catalog — single source of truth for costume options.
 * Each entry: { id, label, emoji, color, promptFragment (≤20 words) }
 */
const costumeCatalog = [
  {
    id: 'knight_armor',
    label: 'Knight Armor',
    emoji: '⚔️',
    color: '#6366f1',
    promptFragment: 'wearing gleaming silver knight armor with a dragon crest shield',
  },
  {
    id: 'space_suit',
    label: 'Space Suit',
    emoji: '🚀',
    color: '#0ea5e9',
    promptFragment: 'wearing a shiny white space suit with a glowing helmet visor',
  },
  {
    id: 'princess_gown',
    label: 'Princess Gown',
    emoji: '👑',
    color: '#ec4899',
    promptFragment: 'wearing a sparkling pink princess gown with a golden tiara',
  },
  {
    id: 'pirate_outfit',
    label: 'Pirate Outfit',
    emoji: '🏴‍☠️',
    color: '#f59e0b',
    promptFragment: 'wearing a rugged pirate coat with a feathered tricorn hat',
  },
  {
    id: 'superhero_cape',
    label: 'Superhero Cape',
    emoji: '🦸',
    color: '#ef4444',
    promptFragment: 'wearing a bold superhero suit with a flowing red cape',
  },
  {
    id: 'wizard_robe',
    label: 'Wizard Robe',
    emoji: '🧙',
    color: '#8b5cf6',
    promptFragment: 'wearing a deep purple wizard robe covered in glowing star patterns',
  },
  {
    id: 'explorer_gear',
    label: 'Explorer Gear',
    emoji: '🧭',
    color: '#22c55e',
    promptFragment: 'wearing rugged explorer gear with a leather satchel and compass',
  },
  {
    id: 'fairy_wings',
    label: 'Fairy Wings',
    emoji: '🧚',
    color: '#a855f7',
    promptFragment: 'wearing a shimmering fairy outfit with translucent rainbow wings',
  },
];

export default costumeCatalog;

File: backend/app/data/costume_catalog.py
"""
Backend mirror of the frontend costume catalog.

Used for:
1. Validating costume IDs on the PUT /api/costume endpoint
2. Looking up prompt_fragment when the frontend only sends the costume ID
"""

COSTUME_CATALOG: dict[str, dict] = {
    "knight_armor": {
        "label": "Knight Armor",
        "emoji": "⚔️",
        "prompt_fragment": "wearing gleaming silver knight armor with a dragon crest shield",
    },
    "space_suit": {
        "label": "Space Suit",
        "emoji": "🚀",
        "prompt_fragment": "wearing a shiny white space suit with a glowing helmet visor",
    },
    "princess_gown": {
        "label": "Princess Gown",
        "emoji": "👑",
        "prompt_fragment": "wearing a sparkling pink princess gown with a golden tiara",
    },
    "pirate_outfit": {
        "label": "Pirate Outfit",
        "emoji": "🏴‍☠️",
        "prompt_fragment": "wearing a rugged pirate coat with a feathered tricorn hat",
    },
    "superhero_cape": {
        "label": "Superhero Cape",
        "emoji": "🦸",
        "prompt_fragment": "wearing a bold superhero suit with a flowing red cape",
    },
    "wizard_robe": {
        "label": "Wizard Robe",
        "emoji": "🧙",
        "prompt_fragment": "wearing a deep purple wizard robe covered in glowing star patterns",
    },
    "explorer_gear": {
        "label": "Explorer Gear",
        "emoji": "🧭",
        "prompt_fragment": "wearing rugged explorer gear with a leather satchel and compass",
    },
    "fairy_wings": {
        "label": "Fairy Wings",
        "emoji": "🧚",
        "prompt_fragment": "wearing a shimmering fairy outfit with translucent rainbow wings",
    },
}

_DEFAULT_PROMPT = "wearing adventure clothes"


def get_costume_prompt(costume_id: str | None) -> str:
    """Return the prompt fragment for a costume ID, or the default."""
    if costume_id and costume_id in COSTUME_CATALOG:
        return COSTUME_CATALOG[costume_id]["prompt_fragment"]
    return _DEFAULT_PROMPT


def is_valid_costume(costume_id: str) -> bool:
    """Return True if the costume ID exists in the catalog."""
    return costume_id in COSTUME_CATALOG
